src/pages/preferences.vue

```diff
@@ -0,0 +1,125 @@
+      </h2>
+      <p>
+        {{ $t(`pref-page.${isSwitchable ? '' : 'non-'}switchable.desc`) }}
+      </p>
+      <ul class="!ps-0">
+        <template v-for="(feature, name) in flags">
+          <li
+            v-if="(feature.status === SWITCHABLE) === isSwitchable"
+            :key="name"
+            class="flex flex-row items-center"
+          >
+            <VCheckbox
+              :id="name"
+              class="flex-row items-center"
+              :checked="featureState(name) === ON"
+              :disabled="!isSwitchable"
+              @change="handleChange"
+            >
+              <div>
+                <strong>{{ name }}</strong>
+                <br />
+                {{ feature.description }}
+              </div>
+            </VCheckbox>
+          </li>
+        </template>
+      </ul>
+    </div>
+
+    <h2>{{ $t('pref-page.store-state') }}</h2>
+    <pre><code>{{ flags }}</code></pre>
+
+    <h2>{{ $t('pref-page.content-filtering') }}</h2>
+    <ul>
+      <template v-for="(_, featName) in flags">
+        <template v-for="featState in FEATURE_STATES">
+          <i18n
+            v-if="featureState(featName) === featState"
+            :key="`${featName}-${featState}`"
+            path="pref-page.explanation"
+            tag="li"
+          >
+            <template #feat-name
+              ><code>{{ featName }}</code></template
+            >
+            <template #feat-state
+              ><code>{{ featState }}</code></template
+            >
+          </i18n>
+        </template>
+      </template>
+      <i18n
+        v-if="featureState('feat_nonexistent') === ON"
+        path="pref-page.explanation"
+        tag="li"
+      >
+        <template #feat-name
+          ><code>{{ $t('feat_nonexistent') }}</code></template
+        >
+        <template #feat-state
+          ><code>{{ $t(ON) }}</code></template
+        >
+      </i18n>
+    </ul>
+  </VContentPage>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, useContext } from '@nuxtjs/composition-api'
+
+import { useFeatureFlagStore } from '~/stores/feature-flag'
+import { SWITCHABLE, ON, OFF, FEATURE_STATES } from '~/constants/feature-flag'
+
+import VContentPage from '~/components/VContentPage.vue'
+import VCheckbox from '~/components/VCheckbox/VCheckbox.vue'
+
+export default defineComponent({
+  name: 'VPreferences',
+  components: {
+    VContentPage,
+    VCheckbox,
+  },
+  setup() {
+    const { app } = useContext()
+    const featureFlagStore = useFeatureFlagStore()
+
+    const flags = computed(() => featureFlagStore.flags)
+
+    /**
+     * Toggle the state of the switchable flag to the preferred value.
+     * @param name
+     * @param checked
+     */
+    const handleChange = ({
+      name,
+      checked,
+    }: {
+      name: string
+      checked: boolean
+    }) => {
+      featureFlagStore.toggleFeature(name, checked ? ON : OFF)
+      app.$cookies.set('features', featureFlagStore.flagStateMap)
```

Comment:
Is there a scenario where we would want to toggle a feature but not set the cookie?
If not, can we move setting the cookie to the store so that components don't have to know about this implementation detail?

Reply:
Ideally components aren't toggle feature flags, only the preferences page is doing so.

Dhruv and I discussed the same separation of concerns question in one of the comments on [this review](https://github.com/WordPress/openverse-frontend/pull/1270#pullrequestreview-941984689) (I can't find the exact comment right now because they won't load for me at the moment).

I tend to agree with you that I'd like the store to encapsulate all the behavior including reading and writing the cookies, but Dhruv's reasons for not doing so also made sense at time, IIRC trying to separate the concerns :slightly_smiling_face:

Reply:
Oh, I missed the fact that a feature flag can ONLY be updated by the preferences page.
I found [the discussion](https://github.com/WordPress/openverse-frontend/pull/1270#discussion_r850293918) you've had:

> The store doesn't read cookie directly it only reads the JSON (which is read from cookies in middleware). The store also doesn't write cookies, it just returns an object (which is written to the cookie in a Vue component).

And still wasn't convinced that not updating the cookie in the store is the best course of action. But since it's only the preferences page, it makes sense.

Reply:
The other issue was that `app.$cookies` would not be available inside the store. I'd defintitely move it to the store (the reasons in favour of doing so are rather strong) if there is a good way to get `app.$cookies` inside the store.